### ISSUE TITLE
A broken or missing config file only prints a message; linting continues with defaults

### ISSUE THREAD

**Issue by raifdmueller** [docToolchain/asciidoc-linter#59]
## Problem

If the file passed with `--config` cannot be read or parsed, the linter prints a message and continues with the default rule set. The exit code reflects only the findings, not the configuration error. In CI this goes unnoticed: rules the user disabled are silently enabled again, and severities fall back to their defaults.

## Minimal reproduction

At commit `52e7fa5982d980eefd5b548f68ff6f58767fe772`, `bad.yml`:

```yaml
rules: [unclosed
```

`file.adoc`:

```asciidoc
= T

text
```

```
asciidoc-linter --config bad.yml --format plain file.adoc; echo $?
```

The same happens with a path that does not exist (`--config missing.yml`).

## Expected vs. actual

- Expected: the linter stops with a clear message on stderr and a distinct non-zero exit code (for example 2, the argparse convention for usage errors).
- Actual: "Error loading config file: ..." is printed to stdout, linting runs with defaults, and the exit code is 0 for a clean document.

## Cause

`asciidoc_linter/linter.py:65-72` — `load_config()` wraps loading and applying the config in `except Exception as e: print(f"Error loading config file: {e}")` and returns normally.

## Proposal

Let `load_config()` raise a dedicated exception (or re-raise), catch it in `cli.main()`, write the message to stderr and return exit code 2. A config file that is empty or lacks a `rules` key can stay valid.

**Comment by rdmueller** [comment]
## Analysis

Confirmed. `AsciiDocLinter.load_config()` (`asciidoc_linter/linter.py`) catches every `Exception`, prints to stdout and returns normally. A broken or missing config therefore silently falls back to defaults, and the exit code depends only on the findings.

**Fix plan (fix order: 5 of 9, independent; do it before #58, because both touch `cli.main()`)**
- New exception `ConfigError` in `linter.py`. `load_config()` raises it for: missing/unreadable file, YAML syntax error, top level not a mapping, `rules` not a mapping, invalid severity value.
- An empty file or a file without a `rules` key stays valid (as the issue proposes). Today `yaml.safe_load` on an empty file returns `None`, and `apply_config(None)` crashes.
- `cli.main()` catches `ConfigError`, writes `Error loading config file: …` to **stderr** and returns **2**.
- Tests: missing file, YAML syntax error, invalid severity, empty file, CLI exit code 2 with the message on stderr.

---
_Generated by [Claude Code](https://claude.ai/code)_
